### PR TITLE
NAVIGATION-170

### PR DESCRIPTION
**Pull request by embrown**
Created menu directive, a helper for the custom navigation directive.

### CODE REVIEW

File: src/directives/bossy.navigation.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                scope.menu = {
+                    open: false,
+                    root: null,
+                    cur: null,
+                };
+
+                $data.getData(attrs.data).then(function(result) {
+                    // Grabbing the first menu object for testing. 
+                    scope.menu.cur = scope.menu.root = new Node(null, result.navigation[0]);
+                    console.log(result.navigation[0]);
```
Comment (radfahrer) [inline src/directives/bossy.navigation.js]: Remove all instances of console.log, console object is not available in all browsers, and should only be use for development purposes.

Comment (embrown) [inline src/directives/bossy.navigation.js]: Fixed. Removed console.log statements.

File: src/bossy.js [hunk 2]
```diff
@@ -19,6 +19,7 @@ var bossy = angular.module('bossy', [
         'bossy.graph',
 		'bossy.input',
 		'bossy.schema',
-		'bossy.tooltip'
+		'bossy.tooltip',
+    'bossy.navigation'
```
Comment (radfahrer) [inline src/bossy.js]: Please clean up the indentation on this file. :-)

File: src/directives/bossy.navigation.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                        for (i = 0; i < menuObj.link.length; i++) {
+                            this.children.push(new Node(this, menuObj.link[i]));
+                        }
+                    }
+                }
+
+                scope.menu = {
+                    open: false,
+                    root: null,
+                    cur: null,
```
Comment (glawson) [inline src/directives/bossy.navigation.js]: What does 'cur' stand for - current?  if so call it current.

**Pull request merged**